test/e2e/case5_policy_automation_test.go

```diff
@@ -135,7 +136,448 @@ var _ = Describe("Test policy automation", func() {
+				)
+				Expect(err).To(BeNil())
+			}
+			By("Waiting until the status on each cluster is updated to Compliant")
+			replicatedPlcList = utils.ListWithTimeout(clientHubDynamic, gvrPolicy, opt, 2, true, defaultTimeoutSeconds)
+			for _, replicatedPlc := range replicatedPlcList.Items {
+				Eventually(func() interface{} {
+					status := replicatedPlc.Object["status"]
+
+					return status.(map[string]interface{})["compliant"]
+				}, 30, 1).Should(Equal(string(policiesv1.Compliant)))
+			}
+
+			By("Should not create any new ansiblejob when Compliant, still four ansiblejobs")
+			Consistently(func() interface{} {
+				ansiblejobList, err := clientHubDynamic.Resource(gvrAnsibleJob).List(
+					context.TODO(), metav1.ListOptions{},
+				)
+				Expect(err).To(BeNil())
+
+				return len(ansiblejobList.Items)
+			}, 15, 1).Should(Equal(4))
+
+			By("Patching policy to make both clusters NonCompliant")
+			opt = metav1.ListOptions{
+				LabelSelector: common.RootPolicyLabel + "=" + testNamespace + "." + case5PolicyName,
+			}
+			replicatedPlcList = utils.ListWithTimeout(clientHubDynamic, gvrPolicy, opt, 2, true, defaultTimeoutSeconds)
+			for _, replicatedPlc := range replicatedPlcList.Items {
+				replicatedPlc.Object["status"] = &policiesv1.PolicyStatus{
+					ComplianceState: policiesv1.NonCompliant,
+				}
+				_, err := clientHubDynamic.Resource(gvrPolicy).Namespace(replicatedPlc.GetNamespace()).UpdateStatus(
+					context.TODO(), &replicatedPlc, metav1.UpdateOptions{},
+				)
+				Expect(err).To(BeNil())
+			}
+			By("Waiting until the status on each cluster is updated to NonCompliant")
+			replicatedPlcList = utils.ListWithTimeout(clientHubDynamic, gvrPolicy, opt, 2, true, defaultTimeoutSeconds)
+			for _, replicatedPlc := range replicatedPlcList.Items {
+				Eventually(func() interface{} {
+					status := replicatedPlc.Object["status"]
+
+					return status.(map[string]interface{})["compliant"]
+				}, 30, 1).Should(Equal(string(policiesv1.NonCompliant)))
+			}
+
+			By("Should not create any new ansiblejob as Non-Compliant, " +
+				"the third event within delayAfterRunSeconds period")
+			Consistently(func() interface{} {
+				ansiblejobList, err := clientHubDynamic.Resource(gvrAnsibleJob).List(
+					context.TODO(), metav1.ListOptions{},
+				)
+				Expect(err).To(BeNil())
+
+				return len(ansiblejobList.Items)
+			}, 15, 1).Should(Equal(4))
+
+			By("After delayAfterRunSeconds is expired, " +
+				"should only create the fifth new ansiblejob for all three events in delayAfterRunSeconds period")
+			Eventually(func() interface{} {
+				ansiblejobList, err := clientHubDynamic.Resource(gvrAnsibleJob).Namespace(testNamespace).List(
+					context.TODO(), metav1.ListOptions{},
+				)
+				Expect(err).To(BeNil())
+				_, err = utils.KubectlWithOutput("get", "ansiblejobs", "-n", testNamespace)
+				Expect(err).Should(BeNil())
+
+				return len(ansiblejobList.Items)
+			}, 240, 1).Should(Equal(5))
```

Comment:
Ideally, you would just patch the clustersWithEvent status to provide an older time for `eventTime` and `automationStartTime` before making the policies non-compliant again so that the test doesn't have to wait 240 seconds.

Reply:
Meet some issues when patching the timestamp. Since it is a test time optimization, I reduced it to 150 seconds and will retry patching in another PR.